README.md

```diff
@@ -109,24 +109,49 @@ To show some simple examples (from
 [cypress/integration/query.spec.js](cypress/integration/query.spec.js) or [cypress/integration/find.spec.js](cypress/integration/find.spec.js)):
 
 ```javascript
-cy.queryByText('Button Text').should('exist')
-cy.queryByText('Non-existing Button Text').should('not.exist')
-cy.queryByLabelText('Label text', {timeout: 7000}).should('exist')
+cy.queryAllByText('Button Text').should('exist')
+cy.queryAllByText('Non-existing Button Text').should('not.exist')
+cy.queryAllByLabelText('Label text', {timeout: 7000}).should('exist')
 cy.findAllByText('Jackie Chan').click({multiple: true})
+
+// findAllByText _inside_ a form element
 cy.get('form').within(() => {
-  cy.findByText('Button Text').should('exist')
+  cy.findAllByText('Button Text').should('exist')
 })
 cy.get('form').then(subject => {
-  cy.findByText('Button Text', {container: subject}).should('exist')
+  cy.findAllByText('Button Text', {container: subject}).should('exist')
 })
+cy.get('form').findAllByText('Button Text').should('exist')
 ```
 
+### Differences DOM Testing Library
+
 `Cypress Testing Library` supports both jQuery elements and DOM nodes. This is
 necessary because Cypress uses jQuery elements, while `DOM Testing Library`
 expects DOM nodes. When you pass a jQuery element as `container`, it will get
 the first DOM node from the collection and use that as the `container` parameter
 for the `DOM Testing Library` functions.
 
+`get*` queries are disabled. `find*` queries do not use the Promise API of
+`DOM Testing Library`, but instead forward to the `get*` queries and use Cypress'
+built-in retryability using error messages from `get*` APIs to forward as error
+messages if a query fails. `query*` also uses `get*` APIs, but disables retryability.
+
+`findBy*` is less useful in Cypress compared to `findAllBy*`. If you intend to limit
```

Comment:
Question:

Does the difference between `*By*` and `*AllBy*` make sense in the context of Cypress? Cypress DOM APIs return a jQuery NodeList with one or more elements. Most of the time this is completely fine and expected, but certain commands expect only one element. For example, `.click` will throw an error if handed multiple elements. `.first()` can easily be used to return only the first one or the previous query can be updated to return only one. If a user expects a certain number (including only one), the command can be followed by `.should('have.length', 1)` if the test is meant to ensure only a single element exists in the DOM and fail otherwise - so single vs multiple is opt-in for other elements.

Reply:
I'm not sure why I would use `findAllBy*` when I expect there to only be one in the first place. Is the appeal just that you can use `findAllBy*` 100% of the time and not worry about how many elements are returned?

I like that `findBy*` communicates (through code) the intent of the test writer (that there should only be one) without having to make an explicit assertion for that.

Reply:
I think the question is around idiomatic Cypress code vs idiomatic @testing-library/* code.

All Cypress commands that return elements (actually jQuery-wrapped NodeLists) return 1 or more elements or fail. The only exception to this is if the command is followed by as `.should('not.exist')`. This is because every command that uses `cy.verifyUpcomingAssertions` implicitly adds `.should('exist')` if no assertions are found. jQuery doesn't fail if it can't find an element by a selector, it just returns an empty node list. Cypress ensures there is at least one element. This combined with the implicit retry mechanism ensures you just tell Cypress what you want and it will retry in the background regardless of application timing. This is what makes Cypress so nice to work with.

Most of the time this is exactly what you expect to happen. It avoids a larger API footprint of ensuring only 1 element. Most of the time there is no issue.

For example:
```ts
cy.contains('button', 'Submit').click()
```

`cy.contains` can return more than 1 element. By default, `cy.click` needs one element and will throw an error saying it expected only one element. If you get an error, you can choose to narrow down the selector. I think Cypress did this so that developers wouldn't be forced to choose between wanting one selector and wanting many - also developers don't have to choose to always do single queries vs many queries and switch when there is a specific case. The Cypress API and errors guide the user into a single API with optional narrowing only when needed.

I'm perfectly happy to remove any documentation suggesting single vs many queries. I just wanted to pose a question - should the API reflect something Cypress users would most likely expect or something non-Cypress users would most expect?

Reply:
The thing I personally find most useful about this library is helpers for common things I need to do. Like selecting an input element by its label.

Perhaps other people value the API to be more strictly adhering to the `@testing-library/dom` API. We've already determined it doesn't make sense in the Cypress context to do that exactly (the `get*` queries are already disabled for example)

Reply:
Cypress does have a mechanism for asserting there should be only one element if necessary:

```ts
cy.get('some-selector').should('have.length', 1)
```

Sometimes the intent is to make sure there is only one

Reply:
I think I'm fine with what you've got here. But I do think we should keep `findBy*` around for the foreseeable future 👍

Reply:
I think I'll make the language more positive.

> But I do think we should keep `findBy*` around for the foreseeable future

Do you mean the current functionality that `findBy*` throws an error if more than one 1 element is found?

Reply:
> Do you mean the current functionality that findBy* throws an error if more than one 1 element is found?

Yes

Reply:
> I think the question is around idiomatic Cypress code vs idiomatic @testing-library/* code.

IMHO idiomatic @testing-library/* code not only provides useful common helpers (generally for selecting elements) _but also encourages best practices_. So, it may deviate from idiomatic code in the library / framework it augments because that library doesn't support/encourage best practice.
 
Consider the example: 
* `cy.findAllBy*(...).first().click()` (More common to Cypress) 
* `cy.findBy*(...).click()` (testing library recommended). 

I would always prefer use the second version, because when my code changes so that selector matches a second element, I expect a test failure on _selecting_ an element, not _clicking the incorrect element_. 

It's possible to avoid this case using:
`cy.findAllBy*('some-selector').should('have.length', 1)` 
But doing that after every `findAllBy*` is tedious, and likely to eventually be forgotten / missed - so in this case I think `findBy*` encourages better testing practices than `findAllBy*` and I don't think @testing-library/*  should shy away from recommending different, better practices than Cypress or any other library it integrates with.

Reply:
Is the update to the readme okay? https://github.com/testing-library/cypress-testing-library/#differences-from-dom-testing-library

Reply:
The update to the readme definitely documents this change 👍 

I'm not sure I agree it should be the recommendation of `@testing-library/*` 👐